### PR TITLE
Switch mysql and postgresql tests to HA Production configs

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

File: examples/external-mysql/run-tests.sh [hunk 1]
```diff
@@ -12,31 +12,35 @@ source "${SCRIPTPATH}/../../.github/scripts/parse-versions.sh"
...
-  kubectl delete ns mysql 2>/dev/null || true
+  kubectl delete ns spire-server 2>/dev/null || true
+  kubectl delete ns spire-system 2>/dev/null || true
 }
 
 trap 'trap - SIGTERM && teardown' SIGINT SIGTERM EXIT
 
-"${helm_install[@]}" mysql mysql --version "$VERSION_MYSQL" --repo "$HELM_REPO_MYSQL" \
-  --namespace mysql \
+kubectl create namespace spire-system --dry-run=client -o yaml | kubectl apply -f -
```
Comment (faisal-memon) [inline examples/external-mysql/run-tests.sh:24]: Why not just `kubectl create namespace spire-system`?

Comment (kfox1111) [reply examples/external-mysql/run-tests.sh:24]: Idempotent. If it exists, it leaves it alone. If it doesn't, it creates it.

File: examples/external-mysql/run-tests.sh [hunk 2]
```diff
@@ -12,31 +12,35 @@ source "${SCRIPTPATH}/../../.github/scripts/parse-versions.sh"
...
+kubectl create namespace spire-server --dry-run=client -o yaml | kubectl apply -f -
+kubectl label namespace spire-server pod-security.kubernetes.io/enforce=restricted || true
+
+helm upgrade --install --create-namespace mysql mysql --version "$VERSION_MYSQL" --repo "$HELM_REPO_MYSQL" \
+  --namespace spire-server \
   --values "${DEPS}/mysql.yaml" \
   --wait
 
-"${helm_install[@]}" --namespace "${ns}" --values "${SCRIPTPATH}/values.yaml" \
+helm upgrade --install --create-namespace --namespace "spire-server" \
```
Comment (faisal-memon) [inline examples/external-mysql/run-tests.sh:34]: The name space is already created above, probably do we need the extra `--create-namespace` here?

Comment (kfox1111) [reply examples/external-mysql/run-tests.sh:34]: yeah, I can remove that.

File: examples/external-mysql/run-tests.sh (hunk 1 above)
Comment (faisal-memon) [reply examples/external-mysql/run-tests.sh:24]: `kubectl create` is not idempotent?

File: examples/external-mysql/run-tests.sh (hunk 2 above)
Comment (kfox1111) [reply examples/external-mysql/run-tests.sh]: Done

File: examples/external-mysql/run-tests.sh (hunk 1 above)
Comment (kfox1111) [reply examples/external-mysql/run-tests.sh:24]: No. it fails if it already exists. :/

```
$ kubectl create namespace foo
namespace/foo created
$ kubectl create namespace foo
Error from server (AlreadyExists): namespaces "foo" already exists
$ echo $?
1
```

create to apply is a common pattern to work around it.

File: examples/external-mysql/run-tests.sh [hunk 3]
```diff
@@ -12,31 +12,35 @@ source "${SCRIPTPATH}/../../.github/scripts/parse-versions.sh"
...
+kubectl label namespace spire-server pod-security.kubernetes.io/enforce=restricted || true
+
+helm upgrade --install mysql mysql --version "$VERSION_MYSQL" --repo "$HELM_REPO_MYSQL" \
+  --namespace spire-server \
   --values "${DEPS}/mysql.yaml" \
   --wait
 
-"${helm_install[@]}" --namespace "${ns}" --values "${SCRIPTPATH}/values.yaml" \
+helm upgrade --install --namespace "spire-server" \
+  --values "${SCRIPTPATH}/values.yaml,${SCRIPTPATH}/../production/values.yaml,${SCRIPTPATH}/../production/values-node-pod-antiaffinity.yaml" \
```
Comment (faisal-memon) [inline examples/external-mysql/run-tests.sh:35]: Previously used spire-system but now you are switching to spire-server. Is spire-system still being used here?

Comment (kfox1111) [reply examples/external-mysql/run-tests.sh:35]: All the privileged system level bits go in spire-system. all the rest of the spire-server bits go into spire-server.

File: examples/external-mysql/run-tests.sh (hunk 1 above)
Comment (faisal-memon) [reply examples/external-mysql/run-tests.sh:24]: Got it, thanks.

File: examples/external-mysql/run-tests.sh (hunk 3 above)
Comment (faisal-memon) [reply examples/external-mysql/run-tests.sh:35]: Youre using the `namespaceOverride` for the multi namespace override. Was missing that part.

**Review by mrsabath — APPROVED** [review]
Nice addition @kfox1111 Thank you! 
LGTM

**Pull request merged**